monkey/monkey_island/cc/main.py

```diff
@@ -34,24 +35,19 @@
 MINIMUM_MONGO_DB_VERSION_REQUIRED = "4.2.0"
 
 
-def main(
-    data_dir,
-    should_setup_only=False,
-    server_config_filename=DEFAULT_SERVER_CONFIG_PATH,
-):
-    logger.info("Starting bootloader server")
+def main(setup_only: bool, server_config_path: str, config_options: IslandConfigOptions):
 
-    env_singleton.initialize_from_file(server_config_filename)
-    initialize_encryptor(data_dir)
-    initialize_services(data_dir)
+    env_singleton.initialize_from_file(server_config_path)
```

Comment:
I don't like that this reads the server config yet again. We have duplicate code for config file reading, which also caused the bug, where if server_config.json doesn't exist, it doesn't get created. If we had a layer of startup arguments, we could pass the environment part of startup arguments here, instead of duplicate code. Also, if we decided to refactor environment params into a separate file, this wouldn't have to change. Now it will have to change, to point to a new file.

Reply:
I agree with you. The environment config is eventually going away. We can talk more about it when I get back. This will get resolved in time.